ideascube/configuration/management/commands/config.py

```diff
@@ -0,0 +1,153 @@
+
+        list = subs.add_parser(
+            'list', help='List configuration namespaces and keys')
+        list.add_argument(
+            'namespace', nargs='?',
+            help='Only list configuration keys for this namespace')
+        list.set_defaults(func=self.list_configs)
+
+        reset = subs.add_parser(
+            'reset', help='Reset a configuration option to its default value')
+        reset.add_argument('namespace', help='The configuration namespace')
+        reset.add_argument('key', help='The configuration key')
+        reset.set_defaults(func=self.reset_config)
+
+        set = subs.add_parser(
+            'set', help='Set the value of a configuration option')
+        set.add_argument('namespace', help='The configuration namespace')
+        set.add_argument('key', help='The configuration key')
+        set.add_argument('value', help='The new value')
+        set.set_defaults(func=self.set_config)
+
+        self.parser = parser
+
+    def handle(self, *_, **options):
+        if 'func' not in options:
+            self.parser.print_help()
+            self.parser.exit(1)
+
+        options['func'](options)
+
+    def _evaluate_value(self, value):
+        if value in ('true', 'false'):
+            # Special case booleans for a better UX
+            value = value.capitalize()
+
+        try:
+            return ast.literal_eval(value)
+
+        except (SyntaxError, ValueError):
+            # Admin probably wanted a string and didn't use enough quotes
+            # examples:
+            #     $ ideascube config set server site-name thename
+            #     $ ideascube config set server site-name 'the name'
+            return str(value)
+
+    def get_config(self, options):
+        namespace = options['namespace']
+        key = options['key']
+
+        try:
+            value = get_config(namespace, key)
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print('%r' % value)
+
+    def help_for_config(self, options):
+        data = {
+            'namespace': options['namespace'],
+            'key': options['key'],
+        }
+
+        try:
+            data.update(get_config_data(data['namespace'], data['key']))
+
+        except (NoSuchConfigurationNamespaceError,
+                NoSuchConfigurationKeyError) as e:
+            raise CommandError(e)
+
+        print(
+            '# {namespace} {key}\n\n'
+            '{summary}\n\n'
+            '* type: {pretty_type}\n'
+            '* default value: {default!r}'.format(**data))
+
+    def list_configs(self, options):
+        namespace = options['namespace']
+
+        if namespace is None:
+            for namespace in get_all_namespaces():
+                print(namespace)
+
+            return
+
+        try:
+            for key in get_namespaced_configs(namespace):
+                print("%s %s" % (namespace, key))
```

Comment:
But you do see how your suggestion would make scripting super hard, because suddenly the script would have to scrape the output and parse it?

It also entirely replaces the `get` command, which suddenly becomes completely useless.

Also, if we just added the summary and type to your suggestion, then it would completely replace the `help` command (which I'm starting to think I should have called `describe`).

Basically, your suggestion implies completely rethinking all the design assumptions I made, so if you want to make changes, then instead of a local suggestion I'd like you to think about the big picture and actually design the whole thing.

I really think that before we start adding some bells and whistles to the command output, we need:

* someone to try and write the scripts that will use this command, as that requires a certain kind of simple output (no scraping/parsing, directly the things needed for the script to continue)
* usage in the field, to see the kind of output that would make more sense to a human and make their job easier

As far as I know, nobody started on the former, so we have no idea what works and what doesn't with the current scripting UX. (and I'd contend that the current input/output would actually work really well for scripts)

You've started on the latter, which is great, but I'd argue that this is too few to really start getting a good feeling of the pain points with the current UX. Can we use it in a few deployments first?

One thing that I think would make a much better UX is tab-completion:

```
$ ideascube config get <tab><tab>
content home server
$ ideascube config get s<tab>
$ ideascube config get server <tab><tab>
site-name
$ ideascube config get server s<tab>
$ ideascube config get server site-name
'Chez Oim'
```

Implementing the above would require a way to list namespaces (and only get namespaces), a way to get keys in a namespace (and only get that).

This is all things I've carefully thought about when designing the UX for this CLI, and so far the local suggestions you and @mgautierfr made would break one thing or another.

Now I'm certainly not saying the UX I designed is perfect, far from it. But it is a global, consistent design based on certain assumptions, and if we're going to change it, I'd like us to instead challenge these very assumptions with sufficient feedback from real-world usage, and then evolve a new global, consistent design.